### ISSUE TITLE
Umbraco 14 RC 3 - Method CreateBlockModel.EnrichBlockItem cannot create a BlockGridModel

### ISSUE THREAD

**Issue by kasparboelkjeldsen** [umbraco/Umbraco-CMS#16227]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

14.0.0-rc3

### Bug summary

The persisted Json from a simple blockgrid without any custom configuration generates json that isn't compatible with the CreateBlockModel -> EnrichBlockItem method in BlockGridPropertyValueCreator.cs

The result is an exception whenever you try to render a blockgrid through MVC or Content Delivery Api

### Specifics

On a completly fresh Umbraco 14 installation with nothing but a page with a blockgrid and a single element in that blockgrid 

This code:

```
@using Umbraco.Cms.Web.Common.PublishedModels;
@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<Page>
@{
    Layout = null;
    var test = Model.Grid;
}
```

Will give the following exception

```
InvalidOperationException: Nullable object must have a value.
System.Nullable<T>.get_Value()
Umbraco.Cms.Core.PropertyEditors.ValueConverters.BlockGridPropertyValueCreator+<>c__DisplayClass3_0.<CreateBlockModel>g__EnrichBlockItem|2(BlockGridItem blockItem, BlockGridLayoutItem layoutItem, BlockGridBlockConfiguration blockConfig, CreateBlockItemModelFromLayout createBlockItem) in BlockGridPropertyValueCreator.cs
+
            blockItem.RowSpan = layoutItem.RowSpan!.Value;

```

To be able to continue my work I've done the following in Umbraco.Cms.Core.PropertyEditors.ValueConverters.BlockGridPropertyValueConverter:

Replaced this line "**blockItem.RowSpan = layoutItem.RowSpan!.Value;**" with "**blockItem.RowSpan = layoutItem.RowSpan ?? 0;**"

![image](https://github.com/umbraco/Umbraco-CMS/assets/137318845/2e960a7a-1807-4538-8901-8ad391b798dd)
I have commented out the offending line and written a  fallback. But I have a feeling this isn't the correct solution. I'm guessing it's on purpose that we expect the model to have a value and that it's an error in how the json for the blockgrid is generated on save.

I'd happily pull request this in, but as noted, I expect that the issue is deeper and that it should be handled at the source instead of having a fallback here.

### Steps to reproduce

Create new Umbraco 14 site with RC3 

Create Document Type with Template called Page - set allow as root node
Create Element Type called Element 
Create Blockgrid on Page, add Element as a block and allow in root.

Create content "Home", add an element to the grid and attempt to output the grid, in any way in MVC.

Ie.

```
@using Umbraco.Cms.Web.Common.PublishedModels;
@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<Page>
@{
    Layout = null;
    var test = Model.Grid;
}
```

### Expected result / actual result

I'd expect to be able to access my blockgrid.

**Comment by nul800sebastiaan** [comment]
I can reproduce this one, unfortunately it also stops the content deliver API cold in it's tracks. 

![image](https://github.com/umbraco/Umbraco-CMS/assets/304656/0990f9a7-9cd2-4737-9736-c0b4b59ce4d9)

I'm guessing it must be some kind of value convertor issue since the backoffice saves and retrieves the values on the node just fine.

**Comment by kasparboelkjeldsen** [comment]
@nul800sebastiaan 
![image](https://github.com/umbraco/Umbraco-CMS/assets/137318845/ba533c76-984f-4003-bbb6-2a2f0f360547)
It looks to me like the persisted propertyData doesn't have any default values for rowSpan

**Comment by kasparboelkjeldsen** [comment]
@nul800sebastiaan sad to report that this issue has returned in 14.1.0-RC, at least when installed by nuget

**Comment by kasparboelkjeldsen** [comment]
@bergmania this is still an issue in 14.1.0-rc2 - notably only when you install from nuget, and not when you build from source code.

**Comment by nielslyngsoe** [comment]
Hi @kasparboelkjeldsen thanks for reporting this issue — I just wanted to confirm it works when building from source code.

We will have to investigate what the difference is between that and the RC version.

**Comment by kasparboelkjeldsen** [comment]
@nielslyngsoe we are definitely at the point where there is a risk I'm just haunted, because I've made this POC git repo https://github.com/kasparboelkjeldsen/umbraco14.1.weirdness which @rickbutterfield  has pulled down and can't replicate, so obviously we are back to it maybe being a user error on my part, but I just replicated it on a home machine that has never run anything Umbraco, so now I don't know what to say.

I'm doing some more testing now to see if I can figure out if I'm haunted or there really is something here, and I apologize in advance if I have wasted anyones time.

**Comment by nielslyngsoe** [comment]
No worries, we should be bullet prof on this, so rather double check so we can avoid having some case that does not work. :-)
I've now tested from source code both 14.1 and 14.2 there it works, I'm going to test from the RC version now. :-)

**Comment by kasparboelkjeldsen** [comment]
@nielslyngsoe ok, no something is definitely up. To eliminate any "my-computer" issue, I've spun up a completly fresh hyper-v Windows 11 dev environment and I can replicate it there 

![image](https://github.com/umbraco/Umbraco-CMS/assets/137318845/5d7870c9-4a5e-41d4-91b5-9982b8bd1272)

**Comment by nielslyngsoe** [comment]
Hi @kasparboelkjeldsen 

I just concluded there is something off with our build, the RC does not contain the files we expected.

Both an issue with files not begin bundles, which we started doing a bit for 14.1, as well we see polyfills for private fields setters, which we should not.

I have handed the task over to the DevOps people.

**Comment by kasparboelkjeldsen** [comment]
@nielslyngsoe great! Glad we caught that!

**Comment by nielslyngsoe** [comment]
Wups, that was my mistake. forget what I wrote above. I had mistakenly gotten a wrong version.

But now I have actually succeed replicating your problem with your project.

It seems like a race condition, it only happens sometimes. But now I have something to investigate.

Thanks

**Comment by kasparboelkjeldsen** [comment]
@nielslyngsoe ok, works (with nuget install) for 14.0.0 and breaks in 14.1.0-rc (and of course rc2) so should be possible to find the difference between those builds

**Comment by nielslyngsoe** [comment]
yeah, well its not that simple cause more features have been build in between these two features.

But I'm on it now, I will give a follow up when I know more :-)

**Comment by nielslyngsoe** [comment]
Okay, I have boiled it down to begin related to `Live editing mode`, if you turn that off you should be fine :-)

**Comment by kasparboelkjeldsen** [comment]
Can confirm - works when live editing mode is disabled

**Comment by nielslyngsoe** [comment]
So it was a race condition 🦥🏎️, and I have made a hotfix:

https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/2101

**Comment by iOvergaard** [comment]
Thanks for your diligence and the nice reproductive repository, @kasparboelkjeldsen.

The bug fix https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/2101 has now been merged and will be included in the final version of 14.1 due next week.
